dvc/repo/experiments/__init__.py

```diff
@@ -51,11 +51,14 @@ class Experiments:
     )
 
     def __init__(self, repo):
-        from dvc.scm import NoSCMError
+        from dvc.scm import NoSCMError, SCMError
 
         if repo.config["core"].get("no_scm", False):
             raise NoSCMError
 
+        if repo.scm.no_commits:
+            raise SCMError("Empty Git repo. Add a commit to use experiments.")
```

Comment:
Let's not raise exception in initializer, please. This should be handled where this exception gets raised.

Reply:
🤔 What if the goal is failing fast? I don't want to queue a bunch of experiments to find out later that they failed because I have an empty repo. And I see it being raised in various points across all `exp` commands, none of which (except `exp init`) work in an empty repo. How does it differ from the `NoSCMError` above?

Reply:
I think @skshetry's point was that `Experiments.__init__()` isn't the right place for it to be raised. Initializing the `dvc.experiments` instance is not limited only to experiments commands, and there's general experiment utility methods that are used outside of experiments commands and should still not fail in an empty git repo.

If we want to make a specific command fail fast in this case, then it should be handled as a part of that command.

Reply:
If we would rather drop this and go in the direction @skshetry suggests to make experiments work in an empty repo, I'm fine with that.

If we plan to leave the implementation as is, this is not a UI tweak for the CLI. AFAIK every underlying functionality of experiments relies on having git history since experiments cannot be generated without it. This is why I asked how it differs from the `NoSCMError` raised in the init.

> If we want to make a specific command fail fast in this case, then it should be handled as a part of that command.

That is what I initially planned, but then it looked like a non-empty repo is a prerequisite for using the class in any capacity. Putting this in the command didn't seem like a good option because:
1. Literally every `dvc exp` command other than `dvc exp init` (and then only without `--run`) fails or does nothing in an empty repo.
2. API methods, and other products like DVCLive that use experiments internals, also benefit from raising the exception here.
3. I'm not sure it would be preferable to be checking the repo for commits inside the command logic?